src/computeSourceMap.ts

```diff
@@ -33,3 +51,60 @@ export default function computeSourceMap(
     names: [],
   };
 }
+
+function computeSourceColumns(code: string, tokens: Array<Token>): Array<number> {
+  const sourceColumns: Array<number> = new Array(tokens.length);
+  let j = 0;
+  let currentMapping = tokens[j].start;
+  let lineStart = 0;
+  for (let i = 0; i < code.length; i++) {
+    if (i === currentMapping) {
+      sourceColumns[j] = currentMapping - lineStart;
+      currentMapping = tokens[++j].start;
+    }
+    if (code.charCodeAt(i) === charCodes.lineFeed) {
+      lineStart = i + 1;
+    }
+  }
+  return sourceColumns;
+}
+
+function computeDetailedSourceMap(
+  code: string,
```

Comment:
Could you call this parameter `generatedCode`? I think that would be more clear and match the `gen-mapping` terminology.